tests/framework/test_time.py

```diff
@@ -510,3 +510,42 @@ def test_step_size_post_processor(builder):
     clock.register_step_modifier(lambda idx: pd.Series(pd.Timedelta(days=0.5), index=idx))
     value = clock._step_size_pipeline(index)
     assert np.all(value == pd.Timedelta(days=2))
+
+
+@pytest.mark.parametrize(
+    "end_day, clock",
+    [
+        (31, "DateTimeClock"),
+        (23, "DateTimeClock"),
+        (31, "SimpleClock"),
+        (23, "SimpleClock"),
```

Comment:
The config of `SimpleClock` looks like this: 
```python
CONFIGURATION_DEFAULTS = {
        "time": {
            "start": 0,
            "end": 100,
            "step_size": 1,
            "standard_step_size": None,
        }
    }
```
Configuring it like this doesn't seem correct